packages/headless/src/api/search/search-api-client.ts

```diff
@@ -141,13 +160,17 @@ export class SearchAPIClient {
     };
   }
 
-  async facetSearch(req: FacetSearchRequest) {
+  async facetSearch(req: FacetSearchRequest): Promise<FacetSearchResponse> {
     const response = await PlatformClient.call({
       ...baseSearchRequest(req, 'POST', 'application/json', '/facet'),
       requestParams: pickNonBaseParams(req),
       ...this.options,
     });
 
+    if (response === PlatformClientErrorResponse.Disconnected) {
+      throw response;
+    }
```

Comment:
For now, I made some functions like this one throw instead of returning because their return type doesn't already expect errors and I felt like this was going out of scope. Those were already throwing beforehand.

Should we handle those now?